intdataplane/endpoint_mgr.go

```diff
@@ -358,9 +367,10 @@ func (m *endpointManager) calculateHostEndpointStatus(id proto.HostEndpointID) (
 }
 
 func (m *endpointManager) resolveWorkloadEndpoints() {
-	// Optimisation, only recalculate the dispatch chains if we've never done so or if the
-	// workloads have changed in some way.
-	needToCheckDispatchChains := m.activeDispatchChains == nil || len(m.pendingWlEpUpdates) > 0
+	if len(m.pendingWlEpUpdates) > 0 {
+		// We're about to make endpoint updates, make sure we re-check the dispatch chains.
```

Comment:
(I'm happy to leave this.  If you do want to fix it, I don't think we need another test cycle before merging!)